### PR TITLE
[MNOE-1041] Capture Organization address on UI

### PR DESCRIPTION
**Pull request by rheasunshine**
@ouranos @adamaziz15

### CODE REVIEW

File: core/config/initializers/00_tenant_config_schema.rb [hunk 1]
```diff
@@ -29,6 +29,16 @@
           description: "Name displayed in the browser title bar and in the public pages",
           default: "My Webstore"
         },
+        organization_requirements: {
+          type: "array",
+          title: "Webstore organization requirements",
+          description: "Information required for new organizations",
+          items: {
+            type: "string",
+            enum: MnoEnterprise::Organization::REQUIRABLE_FIELDS
```
Comment (ouranos) [inline core/config/initializers/00_tenant_config_schema.rb]: Change the constant to be
```ruby
REQUIRABLE_FIELDS = ['main_address']
```
Then you can have:
```ruby
enum: MnoEnterprise::Organization::REQUIRABLE_FIELDS,
'x-schema-form': {
  titleMap: Hash[MnoEnterprise::Organization::REQUIRABLE_FIELDS.map{|f| [f, f.titleize]}]
}
```

so you'll have `main_address` saved in the backend (no longer need to manipulate the string) and `Main Address` in the UI.

File: api/app/views/mno_enterprise/jpi/v1/admin/organizations/_main_address_attributes.json.jbuilder [hunk 2]
```diff
@@ -0,0 +1,5 @@
+json.main_address_attributes do
```
Comment (ouranos) [inline api/app/views/mno_enterprise/jpi/v1/admin/organizations/_main_address_attributes.json.jbuilder]: Should we name this `main_address` and return a `main_address` hash in the JSON (so with the `_attributes` prefix)?

Or is to be consistent with what we sent to mnoe?

File: api/lib/mno_enterprise/concerns/controllers/jpi/v1/admin/organizations_controller.rb [hunk 3]
```diff
@@ -90,7 +92,9 @@ def count
   # POST /mnoe/jpi/v1/admin/organizations
   def create
     # Create new organization
-    @organization = MnoEnterprise::Organization.create!(organization_update_params)
+    @organization = MnoEnterprise::Organization.new(organization_update_params)
+    @organization.save!
```
Comment (ouranos) [inline api/lib/mno_enterprise/concerns/controllers/jpi/v1/admin/organizations_controller.rb]: Why this change?

File: api/lib/mno_enterprise/concerns/controllers/jpi/v1/organizations_controller.rb [hunk 4]
```diff
@@ -28,13 +30,11 @@ def show
...
   def update
-    # Update and Authorize
     authorize! :update, organization
     # Save
-    organization.attributes = organization_update_params
-    changed_attributes = organization.changed_attributes
-    organization.save!
-    MnoEnterprise::EventLogger.info('organization_update', current_user.id, 'Organization update', organization, changed_attributes)
+    organization.update!(organization_update_params)
+    MnoEnterprise::EventLogger.info('organization_update', current_user.id, 'Organization updated', organization)
```
Comment (ouranos) [inline api/lib/mno_enterprise/concerns/controllers/jpi/v1/organizations_controller.rb]: Regression: we no longer save the change diff

File: api/lib/mno_enterprise/concerns/controllers/jpi/v1/organizations_controller.rb [hunk 5]
```diff
@@ -53,7 +53,8 @@ def destroy
   # POST /mnoe/jpi/v1/organizations
   def create
     # Create new organization
-    @organization = MnoEnterprise::Organization.create!(organization_update_params)
+    @organization = MnoEnterprise::Organization.new(organization_update_params)
+    @organization.save!
```
Comment (ouranos) [inline api/lib/mno_enterprise/concerns/controllers/jpi/v1/organizations_controller.rb]: What's the difference?

File: api/lib/mno_enterprise/concerns/controllers/jpi/v1/admin/organizations_controller.rb [hunk 6]
```diff
@@ -211,12 +215,14 @@ def batch_import
...
 
-  def organization_permitted_update_params
-    [:name, :billing_currency]
+  def organization_params
+    params.require(:organization)
   end
 
   def organization_update_params
-    params.fetch(:organization, {}).permit(*organization_permitted_update_params)
+    organization_params.permit(*ATTRIBUTES).tap do |whitelisted|
```
Comment (ouranos) [inline api/lib/mno_enterprise/concerns/controllers/jpi/v1/admin/organizations_controller.rb]: For backward compatibility (as we might have some projects overriding `organization_permitted_update_params`) could we keep:
```ruby
def organization_permitted_update_params
  ATTRIBUTES
end

def organization_update_params
  organization_params.permit(*organization_permitted_update_params).tap do |whitelisted|
    ...
  end
end
```

`ATTRIBUTES` could be renamed to `UPDATEABLE_ATTRIBUTES` (or `WRITEABLE`) for clarity

File: api/lib/mno_enterprise/concerns/controllers/jpi/v1/organizations_controller.rb [hunk 7]
```diff
@@ -154,12 +156,14 @@ def organization
     @organization ||= MnoEnterprise::Organization.find_one(params[:id], DEPENDENCIES)
   end
 
-  def organization_permitted_update_params
-    [:name, :soa_enabled, :industry, :size, :billing_currency]
+  def organization_params
```
Comment (ouranos) [inline api/lib/mno_enterprise/concerns/controllers/jpi/v1/organizations_controller.rb]: See comment on the admin controller

File: core/config/initializers/00_tenant_config_schema.rb [hunk 8]
```diff
@@ -596,20 +606,6 @@
             }
           }
         },
-        tenant_reporting: {
```
Comment (ouranos) [inline core/config/initializers/00_tenant_config_schema.rb]: Did you drop this?

File: api/app/views/mno_enterprise/jpi/v1/admin/organizations/_main_address_attributes.json.jbuilder (hunk 2 above)
Comment (rheasunshine) [reply api/app/views/mno_enterprise/jpi/v1/admin/organizations/_main_address_attributes.json.jbuilder]: This was just for consistency. Should I change?
